includes/multi-currency/Compatibility/WooCommerceProductAddOns.php

```diff
@@ -110,37 +110,64 @@ public function product_addons_params( array $params ): array {
+
+		if ( 0.0 === (float) $addon['price'] ) {
+			$value .= '';
+		} elseif ( 'percentage_based' === $addon['price_type'] && 0.0 === (float) $price ) {
+			$value .= '';
+		} elseif ( 'custom_price' === $addon['field_type'] && $addon['price'] ) {
+			if ( class_exists( 'WC_Product_Addons_Helper' ) ) {
+				$addon_price = wc_price( \WC_Product_Addons_Helper::get_product_addon_price_for_display( $addon['price'], $cart_item['data'] ) );
+				/* translators: %1$s custom addon price in cart */
+				$value           .= sprintf( _x( ' (%1$s)', 'custom price addon price in cart', 'woocommerce-payments' ), $addon_price );
+				$addon['display'] = $value;
+			}
+		} elseif ( 'flat_fee' === $addon['price_type'] && $addon['price'] ) {
+			if ( class_exists( 'WC_Product_Addons_Helper' ) ) {
+				$addon_price = $this->multi_currency->get_price( $addon['price'], 'product' );
+				if ( 'input_multiplier' === $addon['field_type'] ) {
+					// Quantity/multiplier add on needs to be split, calculated, then multiplied by input value.
+					$addon_price = $this->multi_currency->get_price( $addon['price'] / $addon['value'], 'product' ) * $addon['value'];
+				}
+				$addon_price = wc_price( \WC_Product_Addons_Helper::get_product_addon_price_for_display( $addon_price, $cart_item['data'] ) );
+				/* translators: %1$s flat fee addon price in order */
+				$value .= sprintf( _x( ' (+ %1$s)', 'flat fee addon price in cart', 'woocommerce-payments' ), $addon_price );
 			}
-			if ( class_exists( '\WC_Product_Addons_Helper' ) ) {
-				$price = \WC_Product_Addons_Helper::get_product_addon_price_for_display( $price, $cart_item['data'] );
-				$name .= ' (' . wc_price( $price ) . ')';
+		} elseif ( 'quantity_based' === $addon['price_type'] && $addon['price'] && $add_price_to_value ) {
+			if ( class_exists( 'WC_Product_Addons_Helper' ) ) {
+				$addon_price = $this->multi_currency->get_price( $addon['price'], 'product' );
+				if ( 'input_multiplier' === $addon['field_type'] ) {
+					// Quantity/multiplier add on needs to be split, calculated, then multiplied by input value.
+					$addon_price = $this->multi_currency->get_price( $addon['price'] / $addon['value'], 'product' ) * $addon['value'];
+				}
+				$addon_price = wc_price( \WC_Product_Addons_Helper::get_product_addon_price_for_display( $addon_price, $cart_item['data'] ) );
+				/* translators: %1$s addon price in order */
+				$value .= sprintf( _x( ' (%1$s)', 'quantity based addon price in cart', 'woocommerce-payments' ), $addon_price );
 			}
-		} else {
+		} elseif ( 'percentage_based' === $addon['price_type'] && $addon['price'] && $add_price_to_value ) {
 			// Get the percentage cost in the currency in use, and set the meta data on the product that the value was converted.
 			$_product = wc_get_product( $cart_item['product_id'] );
 			$price    = $this->multi_currency->get_price( $price, 'product' );
 			$_product->set_price( $price * ( $addon['price'] / 100 ) );
```

Comment:
Yes, this happens because the percentage amount needs to be calculated on top of the converted product price. Flat fees and quantity based add-ons are added on top of the product price and they don't need the product price to make any calculation about the actual add-on value.

Reply:
Thanks for the explanation.